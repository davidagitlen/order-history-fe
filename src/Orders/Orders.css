.orders-container {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 10px 10px;
  height: 100vh;
  width: 100%;
}