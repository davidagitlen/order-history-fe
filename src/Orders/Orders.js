import React from 'react';
import Order from '../Order/Order';
import './Orders.css';

const Orders = ({orderArray, removePurchase}) => {

  const orderList = orderArray.map(order => <Order id={order.id} key={order.id} image={order.img} name={order.name} description={order.description} price={order.price} removePurchase={removePurchase}/>)
  
  return (
    <div className='orders-container'>
      {orderList}
    </div>
  )

}

export default Orders; 