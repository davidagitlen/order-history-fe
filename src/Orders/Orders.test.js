import React from 'react';
import { shallow } from 'enzyme';
import Orders from './Orders';

describe('Orders', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Orders orderArray={[{id: 1, image: 'broken-link.png', name: 'Blib', description: 'blob', price: 100}]}/>) 
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})