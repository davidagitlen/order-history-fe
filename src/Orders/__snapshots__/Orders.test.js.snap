// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot with all data passed in correctly 1`] = `
<div
  className="orders-container"
>
  <Order
    description="blob"
    id={1}
    key="1"
    name="Blib"
    price={100}
  />
</div>
`;
